src/doc/trpl/ffi.md

```diff
@@ -541,3 +541,26 @@ with one of the non-nullable types, it is represented as a single pointer,
 and the non-data variant is represented as the null pointer. So
 `Option<extern "C" fn(c_int) -> c_int>` is how one represents a nullable
 function pointer using the C ABI.
+
+# Calling Rust code from C
+
+You may wish to compile Rust code in a way so that it can be called from C. This is
+fairly easy, but requires a few things:
+
+```
+#[no_mangle]
+pub extern fn hello_rust() -> *const u8 {
+    "Hello, world!\0".as_ptr()
+}
+```
+
+The `extern "C"` makes this function adhere to the C calling convention, as
+discussed above in "[Foreign Calling
+Conventions](guide-ffi.html#foreign-calling-conventions)". The `no_mangle`
+attribute turns off Rust's name mangling, so that it is easier to link to.
+
+A note about Rust functions intended to be called from C: they _must_
+catch any panics, and then translate their failure to something to
+return to the caller, or abort if that's not possible. It's undefined
+behavior for Rust to unwind into C code, and vice-versa.
```

Comment:
Hm, this is a little in flux now with https://github.com/rust-lang/rust/pull/21186, not sure how much we want to commit to documentation just yet? Could this note be added afterwards?